### DISCUSSION TITLE
Release tarball with submodules

### DISCUSSION THREAD

**Discussion by AngryPenguinPL** [johnfactotum/foliate#1133] [Ideas]
Could you please in release tab include also tarball with submodules included? This would greatly help distributions package this applications. In this case, using git submodules is probably not advisable and will only cause problems in the long run.
In that case git is more of a manual compilation than a system repository. For distro, we need smth that should be permanent, e.g. an immutable tarball, which will offer reproductive builds - especially when many distributions block all access to the Internet during official compilation.

**Comment by johnfactotum** [comment]
> For distro, we need smth that should be permanent, e.g. an immutable tarball, which will offer reproductive builds

Using git would be better than me manually making a tarball in this regard because you can more easily verify that the result is what you want.

> especially when many distributions block all access to the Internet during official compilation.

That shouldn't be an issue as submodules are downloaded before, and not during, any build or compilation step.
